src/transport/raw/GroupAddress.h

```diff
@@ -0,0 +1,66 @@
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "PeerAddress.h"
+#include <app/util/basic-types.h>
+
+namespace chip {
+namespace Transport {
+
+/**
+ * Describes how a group on a Matter network can be addressed.
+ * See 2.5.6.2. IPv6 Multicast Address
+ *
+ * NOTE: The Fabric ID is used directly, without any scoping. Therefore, these addresses
+ * do not fully identify which Fabric they come from, since the root of trust identifier
+ * is not used, unlike in compressed fabric references used for operational discovery.
+ * This means the same GroupAddress may alias between fabrics, but this is OK since the
+ * goal is only to filter traffic to subscribed groups. Group key security ensures that
+ * even if you receive a message from a different fabric due to FabricID aliasing, it
+ * will be ignored or rejected by non-members.
+ */
+class GroupAddress : public PeerAddress
+{
+public:
+    GroupAddress(chip::FabricId fabric, chip::GroupId group) : PeerAddress(Type::kTcp)
+    {
+        const uint8_t raw_address[16] = { 0xff, 0x35, // 0xFF3 (constant: RFC 3306) | 0x5 (scope: Site-Local)
+                                          0x00,       // Reserved
+                                          0x40,       // plen (64-bit long network prefix field)
+                                          0xfd,       // locally assigned ULA prefix (RFC 4193, Section 3.1)
+                                          // The upper 56-bits of the Fabric ID in big-endian order
+                                          static_cast<uint8_t>((fabric >> 56) & 0xff), static_cast<uint8_t>((fabric >> 48) & 0xff),
+                                          static_cast<uint8_t>((fabric >> 40) & 0xff), static_cast<uint8_t>((fabric >> 32) & 0xff),
+                                          static_cast<uint8_t>((fabric >> 24) & 0xff), static_cast<uint8_t>((fabric >> 16) & 0xff),
+                                          static_cast<uint8_t>((fabric >> 8) & 0xff),
+                                          // The lower 8-bits of the Fabric ID
+                                          static_cast<uint8_t>(fabric & 0xff),
+                                          0x00, // Constant
+                                          // 16-bits are the Group Identifier
+                                          static_cast<uint8_t>((group >> 8) & 0xff), static_cast<uint8_t>(group & 0xff) };
+        const uint8_t * p             = raw_address;
+        Inet::IPAddress ipaddr;
+
+        Inet::IPAddress::ReadAddress(const_cast<const uint8_t *&>(p), ipaddr);
```

Comment:
Why is this `const_cast` needed?  It should not be needed.

Reply:
Not needed. Removed.